### PR TITLE
Add CoC, and add examples and better documentation to the readme

### LINKED ISSUE #92: readme.md is less informative about the project
Can you tell me funtionality i can create a codeOfconduct.md and update readme.md for contributors so they can understand about your project more easily also if you can assign me this issue please add label of hacktoberfest and gssoc im new just looking to do some contributions

### PR DESCRIPTION
**Pull request by Anuraj-IND**
sorry for 3 days late response actually im having exams too well see if readme file is good now also thanks for letting me contribute ,if its alright dont forget to add label of hacktoberfest before merging on it to main code file ill also do the codeofconduct file after exams really sorry for late response  
Screenshot 2024-10-27 001927
Screenshot 2024-10-27 001943
Screenshot 2024-10-27 001954

Fixes #92

### CODE REVIEW

**Comment by Anuraj-IND** [comment]
@WyriHaximus respond to it asap if you get time ofc thanks for giving me chance to contribute

**Comment by Anuraj-IND** [comment]
added codeofconduct too @WyriHaximus dont forget to add hacktoberfest label here

**Comment by Anuraj-IND** [comment]
@WyriHaximus please see and respond to it sir

File: README.md
Comment (WyriHaximus) [inline README.md:25]:
```suggestion
```bash
git clone https://github.com/your-repo/php-cpu-core-detection.git
```
```

File: README.md
Comment (WyriHaximus) [inline README.md:22]:
```suggestion
## Clone this repository:
```

File: README.md
Comment (WyriHaximus) [inline README.md:24]:
```suggestion
git clone https://github.com/your-user/php-cpu-core-detection.git
```

File: README.md
Comment (WyriHaximus) [inline README.md:39]: Can you put the core (everything except use statements and inclusion of the autoloader) of the example with a link to it instead of just the link?

File: README.md
Comment (WyriHaximus) [inline README.md:1]:
```suggestion
# PHP CPU Core Count Detection
```

**Review by WyriHaximus — CHANGES_REQUESTED** [review]
Looking good so far, made some suggestions

File: README.md
Comment (Anuraj-IND) [reply README.md]: if i will do 
```bash 
git clone https://github.com/your-repo/php-cpu-core-detection.git
```
but i remove last 3 '```' then it will be like 
```bash
git clone https://github.com/your-repo/php-cpu-core-detection.git 

other data will engulf in this like this

File: README.md
Comment (Anuraj-IND) [reply README.md]: done

File: README.md
Comment (Anuraj-IND) [reply README.md:38]: idk whats the core of example i made up readme file on guess seems like its fine like this i dk anything about php thats why im adding it like that sorry but can u do that from your side rest i did from my end @WyriHaximus check it sir

File: README.md
Comment (Anuraj-IND) [reply README.md:24]: @WyriHaximus btw why youre changing original repo link this link with ,'user' replaced the word ,'repo' will not work for cloning ig but i did it so nvm

**Comment by Anuraj-IND** [comment]
btw other than that link request you asked for i did the rest ig i cant do that tho i have enabled edit by maintainers so you can edit that coz as far as ik we cant directly go to a block of code through a link in md we can only go to file address cant set mouse pointer so idk never did that plus again idk about php much but i did on guessing lol and ig its looking fine so accept it coz October is about to end and yea again thanks for letting me work with you sir it was a nice experience <3 if you still want to end link i really dk how to do it but you can edit it coz all maintainers can edit my pull request hope thats help

File: README.md
Comment (WyriHaximus) [reply README.md:39]: @Anuraj-IND Well, you decided to contribute to a repository for a language you don't really know, which isn't ideal, but I can work with that. Knowing the language isn't perse a requirement, but would have been helpful. What will help you further down is recognizing different techniques/patterns regardless of the language. In PHP's case these lines `"include"` which classes/functions we use: https://github.com/WyriHaximus/php-cpu-core-detector/blob/master/examples/async.php#L3-L5 while this one loads the autoloader: https://github.com/WyriHaximus/php-cpu-core-detector/blob/master/examples/async.php#L7

So for this package the core of the examples would be https://github.com/WyriHaximus/php-cpu-core-detector/blob/master/examples/async.php#L9-L20 and https://github.com/WyriHaximus/php-cpu-core-detector/blob/master/examples/sync.php#L8-L15

**Comment by Anuraj-IND** [comment]
What I have to do now just tell me except that linking thing I can't do
that atm

**Comment by Anuraj-IND** [comment]
So which part I had to link ig. Oe I'm understanding a bit you told me to
link only main parts so what main part of code block you were referring to
again tell me that and ig then work will done

**Comment by Anuraj-IND** [comment]
Thanks I just saw your message completely actually I was traveling and I
was having issue on mail and my GitHub app wasn't working properly so I'll
start working on it shortly really thanks for guiding me 🙇

**Comment by WyriHaximus** [comment]
> October is about to end

No worries, no stress, this is exactly why I linked directly to the lines. In the future the lines are also at the end of the URL in the hash section.

The final thing I want to do is squash everything into a single commit. But we'll get to that once everything is good. Having a look at your latest changes now.

File: README.md
Comment (Anuraj-IND) [reply README.md]: done @WyriHaximus thanks again because of you i was able to learn something new in markdown i.e. how to link properly thanks i merged it in my pull request

**Comment by Anuraj-IND** [comment]
Um why some of my integrations are generating error I just changed the link
as you asked me so why it's generating error can you tell me

**Comment by Anuraj-IND** [comment]
> > October is about to end
> 
> No worries, no stress, this is exactly why I linked directly to the lines. In the future the lines are also at the end of the URL in the hash section.
> 
> The final thing I want to do is squash everything into a single commit. But we'll get to that once everything is good. Having a look at your latest changes now.

@WyriHaximus why im getting mail of continuous integration failure did i do something wrong i just edited those two lines and added link to main code block in the async and sync examples

File: README.md
Comment (WyriHaximus) [inline README.md:39]: Meant like this instead of only linking to them:

### Async

```php
~~The code on the linked lines here~~
```
[Full Example](examples/async.php)

### Sync

```php
~~The code on the linked lines here~~
```
[Full Example](examples/sync.php)

Comment (Anuraj-IND) [reply README.md:39]: oh lemme try

**Comment by WyriHaximus** [comment]
Don't worry about the integrations. Those aren't relevant to your changes. Need to update this package anyone once this PR is in.

**Comment by Anuraj-IND** [comment]
> Don't worry about the integrations. Those aren't relevant to your changes. Need to update this package anyone once this PR is in.

@WyriHaximus check it now is it fine ?

**Comment by WyriHaximus** [comment]
Yes like that 😄 !  Was about to link to a different repository showing off how 😅

**Comment by Anuraj-IND** [comment]
Lol you must be laughing when I just copy pasted your code without reading
it properly 😂😂😂😂 like what a dumb person he is even after telling him
all he's not using his brain 😭😂😂😂 I'm feeling pretty dumb rn

File: README.md
Comment (WyriHaximus) [inline README.md:38]: The only thing I could change here is add `php` at the end of this line and the other example start to enable syntax highlighting

Comment (Anuraj-IND) [reply README.md:38]: which line tell me number ill add php in the end

**Comment by WyriHaximus** [comment]
All looks good, before I merge it can you squash it all in one commit? You can use rebase to do that, have you ever done that?

**Comment by Anuraj-IND** [comment]
> All looks good, before I merge it can you squash it all in one commit? You can use rebase to do that, have you ever done that?

tbh i have done it through git bash once in my life in past in old laptop

**Comment by WyriHaximus** [comment]
> > All looks good, before I merge it can you squash it all in one commit? You can use rebase to do that, have you ever done that?
> 
> tbh i have done it through git bash once in my life in past in old laptop

That's fine, going for lunch now, but when I get back I'll get you the exact commands  to run 👍 .

**Comment by Anuraj-IND** [comment]
> > > All looks good, before I merge it can you squash it all in one commit? You can use rebase to do that, have you ever done that?
> > 
> > 
> > tbh i have done it through git bash once in my life in past in old laptop
> 
> That's fine, going for lunch now, but when I get back I'll get you the exact commands to run 👍 .

oki oki

**Comment by Anuraj-IND** [comment]
Screenshot 2024-10-29 171651
Screenshot 2024-10-29 171726
im stuck here after replacing all of my commits with s for squash i stuck coz my pointer isnt going down from that last line so im not able to type wq! cmd hence you need to tell me what im doing wrong here @WyriHaximus so eat up fast lol

**Comment by WyriHaximus** [comment]
Only use squash for everyone but the first commit so you squash them into that one.

**Comment by Anuraj-IND** [comment]
> Only use squash for everyone but the first commit so you squash them into that one.

tell me your way idk how to do it tell me step by step

**Comment by WyriHaximus** [comment]
Since you are PRing from the same branch as you're PRing to well start off with adding this repo as the upstream remote:

```shell
git remote add upstream https://github.com/WyriHaximus/php-cpu-core-detector
```

Make sure to fetch it:

```shell
git fetch upstream
```

Then:

```shell
git rebase -i upstream/master
```

And change everything to this, the `r` for the first one is because we're going to make it a little nicer:

image

On the first screen you're getting change the commit title to:

```
Add CoC and add examples and better documentation to the readme
```

On the first conflict change:

image

To:

image

Add `README.md`:

```shell
git add README.md
```

Run:

```shell
git rebase --continue
```

On the second conflict change this:

image

To this:

image

Add `README.md`:

```shell
git add README.md
```

Run:

```shell
git rebase --continue
```

When it's all good, run:

```shell
git push --force-with-lease
```

**Comment by Anuraj-IND** [comment]
@WyriHaximus problem isnt that i was doing the same thing too i was only going to squash my commits tho i recalled last 20 commits just to be sure well problem is this thing 
Screenshot 2024-10-29 180107
this 
Screenshot 2024-10-29 180122
highlighted line i need to move my cursor down im trying but its not working so tell me how to do that

**Comment by Anuraj-IND** [comment]
Screenshot 2024-10-29 181446
this is what i have rn now what? @WyriHaximus

**Comment by Anuraj-IND** [comment]
see the terminal and editor window and tell me now what to do im a bit confused
Screenshot 2024-10-29 181910

**Comment by WyriHaximus** [comment]
Not sure why that would be an issue, you can still save it right?

**Comment by Anuraj-IND** [comment]
> Not sure why that would be an issue, you can still save it right?

im not able to do it im trying ik how to rebase it i rebased it 2 times i had to go through all 15 commits :") now idk why im not able to squash my tiny brain cant comprehend such high end activity i should just sink in the shame of nothingness (can you di it from your end?)

**Comment by WyriHaximus** [comment]
If you been through all commits you should be able to push it without any issues. Have you tried closing code and anything else that might mess with the files git creates while doing this?

**Comment by Anuraj-IND** [comment]
> If you been through all commits you should be able to push it without any issues. Have you tried closing code and anything else that might mess with the files git creates while doing this?

@WyriHaximus im able to rebase it i did it as i said problem is after rebase or while rebasing idk what steps i have to perform to squash it in single commit and im getting confused at that point

**Comment by Anuraj-IND** [comment]
> > > If you been through all commits you should be able to push it without any issues. Have you tried closing code and anything else that might mess with the files git creates while doing this?
> > 
> > 
> > @WyriHaximus im able to rebase it i did it as i said problem is after rebase or while rebasing idk what steps i have to perform to squash it in single commit and im getting confused at that point
> 
> The steps I gave you should work as I did them using your branch locally before posting them here to ensure they would as expected. So I'm equally confused why they don't work for you.

its only increasing my commits cant u squash from ur end?

**Comment by WyriHaximus** [comment]
> > > > If you been through all commits you should be able to push it without any issues. Have you tried closing code and anything else that might mess with the files git creates while doing this?
> > > 
> > > 
> > > @WyriHaximus im able to rebase it i did it as i said problem is after rebase or while rebasing idk what steps i have to perform to squash it in single commit and im getting confused at that point
> > 
> > 
> > The steps I gave you should work as I did them using your branch locally before posting them here to ensure they would as expected. So I'm equally confused why they don't work for you.
> 
> its only increasing my commits cant u squash from ur end?

Yeah let me have a look after work 👍

**Comment by Anuraj-IND** [comment]
> > > > > If you been through all commits you should be able to push it without any issues. Have you tried closing code and anything else that might mess with the files git creates while doing this?
> > > > 
> > > > 
> > > > @WyriHaximus im able to rebase it i did it as i said problem is after rebase or while rebasing idk what steps i have to perform to squash it in single commit and im getting confused at that point
> > > 
> > > 
> > > The steps I gave you should work as I did them using your branch locally before posting them here to ensure they would as expected. So I'm equally confused why they don't work for you.
> > 
> > 
> > its only increasing my commits cant u squash from ur end?
> 
> Yeah let me have a look after work 👍

@WyriHaximus 

ig its not rebasing as planned its showing this thing but nothing more like it should be shoulding rebasing progress or something

**Comment by Anuraj-IND** [comment]
> Have you tried O or L to see what happens?

i tried both after typing either of those its disappering and im still on same screen nothing is working i tried in vs code and then in notepad too hopefully in notepad i saw that commit window once and i did see 4 commits i guess you squash rest thanks for that and i tried to just squash that just basic squash nothing else i just did p for first commit and rest will be s so that all will be squashed in first commit readme1 but after saving that when i exit the window it wasnt squashing nor it was showing any error so now im just lost  please do it from ur side sir and accept that pull request if nothing else is there for me to edit thankyou

**Comment by Anuraj-IND** [comment]
@WyriHaximus have you squashed it ?

**Comment by WyriHaximus** [comment]
@Anuraj-IND Chill yes I just did. Not sure how you managed to make it worse tho. Looked like there where lots of left overs from rebasing in there

**Review by WyriHaximus — APPROVED** [review]

**Pull request merged**
